codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesunionstructure.java

```diff
@@ -0,0 +1,3951 @@
+            return allTypesUnionStructure(AllTypesUnionStructure.builder().applyMutation(allTypesUnionStructure).build());
+        }
+    }
+
+    static final class BuilderImpl implements Builder {
+        private String stringMember;
+
+        private Integer integerMember;
+
+        private Boolean booleanMember;
+
+        private Float floatMember;
+
+        private Double doubleMember;
+
+        private Long longMember;
+
+        private Short shortMember;
+
+        private List<String> simpleList = DefaultSdkAutoConstructList.getInstance();
+
+        private List<String> listOfEnums = DefaultSdkAutoConstructList.getInstance();
+
+        private List<Map<String, String>> listOfMaps = DefaultSdkAutoConstructList.getInstance();
+
+        private List<SimpleStruct> listOfStructs = DefaultSdkAutoConstructList.getInstance();
+
+        private List<Map<String, String>> listOfMapOfEnumToString = DefaultSdkAutoConstructList.getInstance();
+
+        private List<Map<String, SimpleStruct>> listOfMapOfStringToStruct = DefaultSdkAutoConstructList.getInstance();
+
+        private Map<String, List<Integer>> mapOfStringToIntegerList = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, String> mapOfStringToString = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, SimpleStruct> mapOfStringToSimpleStruct = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, String> mapOfEnumToEnum = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, String> mapOfEnumToString = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, String> mapOfStringToEnum = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, SimpleStruct> mapOfEnumToSimpleStruct = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, List<String>> mapOfEnumToListOfEnums = DefaultSdkAutoConstructMap.getInstance();
+
+        private Map<String, Map<String, String>> mapOfEnumToMapOfStringToEnum = DefaultSdkAutoConstructMap.getInstance();
+
+        private Instant timestampMember;
+
+        private StructWithTimestamp structWithNestedTimestampMember;
+
+        private SdkBytes blobArg;
+
+        private StructWithNestedBlobType structWithNestedBlob;
+
+        private Map<String, SdkBytes> blobMap = DefaultSdkAutoConstructMap.getInstance();
+
+        private List<SdkBytes> listOfBlobs = DefaultSdkAutoConstructList.getInstance();
+
+        private RecursiveStructType recursiveStruct;
+
+        private BaseType polymorphicTypeWithSubTypes;
+
+        private SubTypeOne polymorphicTypeWithoutSubTypes;
+
+        private String enumType;
+
+        private Underscore_Name_Type underscore_Name_Type;
+
+        private Document myDocument;
+
+        private AllTypesUnionStructure allTypesUnionStructure;
+
+        private Type type = Type.UNKNOWN_TO_SDK_VERSION;
```

Comment:
Not to my knowledge. WDYT?

Reply:
I can't really thing of a good use case for a specific "no value set" `Type` value, but I think intuitively, it might more sense to just set this to `null` if no members are set?

Reply:
For responses, `UNKNOWN_TO_SDK_VERSION` actually makes sense for zero members because it likely means that the service responded with a type that doesn't exist in the union (so we populated nothing into the structure, because we didn't check that new field). Upgrading the SDK would likely fix the problem, because then the SDK would know about the new type.

For requests, it's a bit more murky, but there's no concept of 'request' and 'response' structures, so we can't meaningfully make a distinction.

More than one member being populated I could see being the thing we might want to differentiate on (return null for the type instead of UNKNOWN_TO_SDK_VERSION), but I didn't want the customer to have to worry about the two different cases for responses, because the "more than one" shouldn't happen.

Reply:
Ah good point about forwards compatibility on response; though I think `null` type in that situation would still suffice no? Since `null` would mean none of the members are set.

For clarity I'm advocating for

 - null -> no member set
 - UNKNOWN_TO_SDK_VERSION -> multiple members set

Reply:
We use UNKNOWN_TO_SDK_VERSION in enums already to mean "a member we're not aware of, but upgrading your SDK version might help with that", so we'd probably not want to use the same enum name to mean something different ("multiple members"). 

I like having an enum value to represent "a member we're not aware of", because switching on a null value doesn't work, and we want to ENCOURAGE customers to implement code to handle this case, not the other way around. 

If we want to differentiate, we could do

* UNKNOWN_TO_SDK_VERSION -> no member set
* null -> multiple members set

That would preserve the existing meaning of "UNKNOWN_TO_SDK_VERSION" to be "the SDK doesn't know, but upgrading might help", and "null" to mean "the SDK doesn't know, and upgrading wouldn't help".

Reply:
> I like having an enum value to represent "a member we're not aware of", because switching on a null value doesn't work, and we want to ENCOURAGE customers to implement code to handle this case, not the other way around.

Great point! Yes, completely agree with this.

> That would preserve the existing meaning of "UNKNOWN_TO_SDK_VERSION" to be "the SDK doesn't know, but upgrading might help", and "null" to mean "the SDK doesn't know, and upgrading wouldn't help".

Nice I like this change

> The only other option I can think of would be to let the SDK explicitly tell the union type "the service returned something I don't know" in the unmarshallers to meaningfully set UNKNOWN_TO_SDK_VERSION, and have every other case (0 members on requests, 2+ members on requests, 2+ members on response) have a null type().

The marshaller would need to check for any other members that are not modeled which creates more work done in the marshalling step. I feel like you said, that `UNKNOWN_TO_SDK_VERSION` is a much more efficient solution for that.